CRM/Contribute/Form/Contribution.php

```diff
@@ -1324,6 +1324,12 @@ public function testSubmit($params, $action, $creditCardMode = NULL) {
       ));
       $this->_id = $params['id'];
       $this->_values = $existingContribution;
+      if (CRM_Contribute_BAO_Contribution::checkContributeSettings('invoicing')) {
+        $this->_values['tax_amount'] = civicrm_api3('contribution', 'getvalue', array(
+          'id' => $params['id'],
+          'return' => 'tax_amount',
+        ));
+      }
```

Comment:
@monish I'm curious to know why this is necessary. The old code was referencing this variable already, just as the new code is. Was it sometimes not present?

Reply:
@colemanw for the added unit test it needs default tax_amount information in $this->_values, but unfortunately Contribution.getsingle API doesn't have the value for this attribute, so yes its something not present earlier and there was no UT written, that need tax_amount (until now). Other alternative to deal with it is to set this as one of the default return properties as I did [here](https://github.com/civicrm/civicrm-core/pull/11440/files#diff-0a639f0cf282805e238274495195e6a4R820) But after discussing with Eileen I decided to fetch its value on demand.